File: App/app/src/main/java/com/cinema/ticket_booking/data/model/response/UpcomingShowtimeResponse.java
package com.cinema.ticket_booking.data.model.response;

import com.google.gson.annotations.SerializedName;

public class UpcomingShowtimeResponse {

    @SerializedName("showtimeId")
    public String showtimeId;

    @SerializedName("movieTitle")
    public String movieTitle;

    @SerializedName("moviePosterUrl")
    public String moviePosterUrl;

    @SerializedName("screenName")
    public String screenName;

    @SerializedName("startTime")
    public String startTime; // ISO String, parse ở UI layer

    @SerializedName("minutesUntilStart")
    public long minutesUntilStart;

    @SerializedName("urgency")
    public String urgency; // "SOON" | "UPCOMING"
}

File: App/app/src/main/java/com/cinema/ticket_booking/data/model/response/UpcomingShowtimeResponse.kt
package com.cinema.ticket_booking.data.model.response

import com.google.gson.annotations.SerializedName

data class UpcomingShowtimeResponse(
    @SerializedName("showtimeId")
    @JvmField val showtimeId: String?,

    @SerializedName("movieTitle")
    @JvmField val movieTitle: String?,

    @SerializedName("moviePosterUrl")
    @JvmField val moviePosterUrl: String?,

    @SerializedName("screenName")
    @JvmField val screenName: String?,

    @SerializedName("startTime")
    @JvmField val startTime: String?,

    @SerializedName("minutesUntilStart")
    @JvmField val minutesUntilStart: Long,

    @SerializedName("urgency")
    @JvmField val urgency: String?
)
